metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
I'll be honest, the Java 8 defaults weren't obvious to me either until I looked closer at the MessageParser interface :)

All this synchronization talk makes me nervous.  Would it make things simpler if we detected the config change in ParserBolt.execute and re-initialized the grok object from there?  We could save a cached copy of the config in the ParserBolt on each call to execute and detect changes before parse is called using the hashing approach you suggested earlier.  This is the reason I initially changed the interface parse method to include the most recent config, to avoid multi-threading complexity.  This is approach is very similar, we're just using a different method to pass this config in and detecting a change in ParserBolt instead of GrokParser.  The parse method is synchronous by nature anyways.  It doesn't need to be aware of a config change immediately, only when the parse method is called.